server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/model/Keyword.scala

```diff
@@ -69,21 +52,23 @@ object Keyword {
       | must not contain characters with hex from '\\u0000' to '\\u00019'
       | or {'(' ')' '{' ']' '%' '*' '\"' '\\'} """.stripMargin
 
-  def parse(flagName: String): Option[Keyword] = {
-    if (isValid(flagName)) return Some(new Keyword(flagName))
-    throw new IllegalArgumentException(VALIDATION_MESSAGE)
+  def parse(flagName: String): Either[String, Keyword] = {
+    if (isValid(flagName)) Right(new Keyword(flagName))
+    Left(VALIDATION_MESSAGE)
   }
 
-  def of(flagName: String): Keyword = parse(flagName).get
+  def of(flagName: String): Keyword = parse(flagName) match {
+    case Left(errorMessage: String) => throw new IllegalArgumentException(errorMessage)
+    case Right(keyword: Keyword) => keyword
+  }
 
   def fromFlag(flag: Flags.Flag): Option[Keyword] = Some(IMAP_SYSTEM_FLAGS.get(flag))
 
-  private def isValid(flagName: String): Boolean = {
-    if (StringUtils.isBlank(flagName)) return false
-    if (flagName.length < FLAG_NAME_MIN_LENGTH || flagName.length > FLAG_NAME_MAX_LENGTH) return false
-    if (!FLAG_NAME_PATTERN.containsAll(flagName)) return false
-    true
-  }
+  def isValid(flagName: String): Boolean = flagName match {
+      case _ if StringUtils.isBlank(flagName) => false
+      case _ if flagName.length < FLAG_NAME_MIN_LENGTH || flagName.length > FLAG_NAME_MAX_LENGTH => false
+      case _ => FLAG_NAME_PATTERN.containsAll(flagName)
+    }
 }
 
 class Keyword(val flagName: String) {
```

Comment:
You removed equals/hashcode but you forgot to make the Keyword class a case class